x/leverage/keeper/keeper_test.go

```diff
@@ -128,6 +147,314 @@ func (suite *IntegrationTestSuite) TestWithdrawAsset_Valid() {
 	suite.Require().Equal(int64(0), uTokenBalance.Amount.Int64())
 }
 
+// Helper function: Initialize the common starting scenario from which borrow and repay tests stem:
+func (suite *IntegrationTestSuite) initBorrowScenario() (lender, bum sdk.AccAddress) {
+	app, ctx := suite.app, suite.ctx
+	// register uumee and u/uumee as an accepted asset+utoken pair
+	app.LeverageKeeper.SetTokenDenom(ctx, umeeapp.BondDenom)
+	// create an account and address which will represent a lender
+	lenderAddr := sdk.AccAddress([]byte("addr______________00"))
+	lenderAcc := app.AccountKeeper.NewAccountWithAddress(ctx, lenderAddr)
+	app.AccountKeeper.SetAccount(ctx, lenderAcc)
+	// create an account and address which will represent a user with no assets
+	bumAddr := sdk.AccAddress([]byte("addr______________02"))
+	bumAcc := app.AccountKeeper.NewAccountWithAddress(ctx, bumAddr)
+	app.AccountKeeper.SetAccount(ctx, bumAcc)
+	// mint and send 10k umee to lender
+	suite.Require().NoError(app.BankKeeper.MintCoins(ctx, minttypes.ModuleName, tCoins("umee", 10000)))
+	suite.Require().NoError(app.BankKeeper.SendCoinsFromModuleToAccount(ctx, minttypes.ModuleName, lenderAddr, tCoins("umee", 10000)))
+	// lender lends 1k umee and receives 1k u/umee
+	err := app.LeverageKeeper.LendAsset(ctx, lenderAddr, tCoin("umee", 1000))
+	suite.Require().NoError(err)
+	// return the three account addresses
+	return lenderAddr, bumAddr
+	// The starting scenario is thus:
+	// - umee and u/umee are accepted assets
+	// - a "lender" user has 9k umee and 1k u/umee
+	// - the leverage module has 1k umee due to lender's lending
+	// - a "bum" user has an address but no assets
+}
+
+func (suite *IntegrationTestSuite) TestBorrowAsset_Invalid() {
+	lenderAddr, _ := suite.initBorrowScenario() // create initial conditions
+	app, ctx := suite.app, suite.ctx            // get ctx after init
```

Comment:
nit: please avoid redundant comments 👍 
```suggestion
	lenderAddr, _ := suite.initBorrowScenario()
	app, ctx := suite.app, suite.ctx
```